Sources/Subprocess/AsyncLineSequence.swift

```diff
@@ -0,0 +1,167 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+@available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+public struct AsyncLineSequence<Base: AsyncSequence>: AsyncSequence where Base.Element == UInt8 {
+    public typealias Element = String
+
+    var base: Base
+
+    public struct AsyncIterator: AsyncIteratorProtocol {
+        public typealias Element = String
+
+        var byteSource: Base.AsyncIterator
+        var buffer: Array<UInt8> = []
+        var leftover: UInt8? = nil
+
+        internal init(underlyingIterator: Base.AsyncIterator) {
+            byteSource = underlyingIterator
+        }
+
+        // We'd like to reserve flexibility to improve the implementation of
+        // next() in the future, so aren't marking it @inlinable. Manually
+        // specializing for the common source types helps us get back some of
+        // the performance we're leaving on the table.
+        @_specialize(where Base == Subprocess.AsyncBytes)
+        public mutating func next() async rethrows -> String? {
+            /*
+             0D 0A: CR-LF
+             0A | 0B | 0C | 0D: LF, VT, FF, CR
+             E2 80 A8:  U+2028 (LINE SEPARATOR)
+             E2 80 A9:  U+2029 (PARAGRAPH SEPARATOR)
+             */
+            let _CR: UInt8 = 0x0D
+            let _LF: UInt8 = 0x0A
```

Comment:
instead of hardcoding `0x0A` just use `UInt8(ascii: "\n")`. The emitted code is the same but it's much much more readable.

Ideally, even remove all the constants and replace by `UInt8(ascii: ...)`